### PR TITLE
feat: native auth

### PR DESCRIPTION
**Pull request by Nirajn2311**
## TODO

- [x] Loading indicator

### CODE REVIEW

**Comment by Nirajn2311** [comment]
**NOTE** - To be merged after changes in freeCodeCamp/freeCodeCamp#47586 is deployed and tested with the mobile app

**Comment by Nirajn2311** [comment]
**NOTE** - To be merged after changes in freeCodeCamp/freeCodeCamp#47591 is deployed and tested with the mobile app

File: mobile-app/lib/ui/views/learn/challenge/challenge_viewmodel.dart [hunk 1]
```diff
@@ -376,12 +378,32 @@ class ChallengeViewModel extends BaseViewModel {
     }
   }
 
+  void passChallenge(Challenge? challenge) async {
```
Comment (Sembauke) [inline mobile-app/lib/ui/views/learn/challenge/challenge_viewmodel.dart:381]: What does this do?

**Review by Sembauke — DISMISSED** [review]

File: mobile-app/lib/ui/views/learn/challenge/challenge_viewmodel.dart (hunk 1 above)
Comment (Nirajn2311) [reply mobile-app/lib/ui/views/learn/challenge/challenge_viewmodel.dart:381]: That function is for submitting the challenge when the user traverse to the next one. It generates the body and passes it on to the learn service where that'll then send the request to the respective endpoint.

**Review by Sembauke — APPROVED** [review]

**Pull request merged**